oprf-client/src/types.rs

```
@@ -0,0 +1,45 @@
+use eddsa_babyjubjub::{EdDSAPrivateKey, EdDSAPublicKey, EdDSASignature};
+use oprf_core::proof_input_gen::query::MAX_PUBLIC_KEYS;
+use oprf_types::{MerkleEpoch, MerkleRoot, RpId, ShareEpoch, crypto::UserPublicKeyBatch};
+
+pub struct CredentialsSignature {
+    // Credential Signature
+    pub type_id: ark_babyjubjub::Fq,
+    pub issuer: EdDSAPublicKey,
+    pub hashes: [ark_babyjubjub::Fq; 2], // [claims_hash, associated_data_hash]
+    pub signature: EdDSASignature,
+    pub genesis_issued_at: u64,
+    pub expires_at: u64,
+}
+
+pub struct MerkleMembership {
+    pub epoch: MerkleEpoch,
+    pub root: MerkleRoot,
+    pub depth: u64,
+    pub mt_index: u64,
+    pub siblings: Vec<ark_babyjubjub::Fq>,
+}
+
+pub struct OprfQuery {
+    pub rp_id: RpId,
+    pub share_epoch: ShareEpoch,
+    pub action: ark_babyjubjub::Fq,
+    pub nonce: ark_babyjubjub::Fq,
+    pub current_time_stamp: u64,
+    pub nonce_signature: k256::ecdsa::Signature,
+}
+
+pub struct UserKeyMaterial {
+    pub pk_batch: UserPublicKeyBatch,
+    pub pk_index: u64, // 0..6
+    pub sk: EdDSAPrivateKey,
+}
+
+impl UserKeyMaterial {
+    pub fn public_key(&self) -> ark_babyjubjub::EdwardsAffine {
+        if self.pk_index >= MAX_PUBLIC_KEYS as u64 {
+            panic!("out-of-bounds");
```

this check + panic is kinda redundant no? if you just access the array and it is out of bounds a meaningful panic happens anyways
we could return a option if oob or just let it panic in the access